botocore/loaders.py

```diff
@@ -107,10 +107,21 @@
 from botocore import BOTOCORE_ROOT
 from botocore.compat import json
 from botocore.compat import OrderedDict
+from botocore.compat import HAS_GZIP
 from botocore.exceptions import DataNotFoundError, UnknownServiceError
 from botocore.utils import deep_merge
 
 
+_JSON_OPEN_METHODS = {
```

Comment:
Very minor nit, but should we have this be more general than JSON? It's just a suffix to open function map.

Reply:
I had it as `_OPEN_METHODS` but it's only used / specific to the `JSONFileLoader` so I decided to rename it. I think if we want to expand this to support more formats + Gzip on top of them we should look at a more serious refactor where support for compression is closer to a mixin that sits on top of the base loader class.